source/FPSciApp.cpp

```diff
@@ -260,21 +260,42 @@ void FPSciApp::updateControls(bool firstSession) {
 	if (startupConfig.waypointEditorMode) { waypointManager->updateControls(); }
 
 	// Update the player controls
-	if(notNull(m_playerControls)) removeWidget(m_playerControls);
+	bool visible = false;
```

Comment:
Are these changes through about line 300 the ones that cause menus to remember their previous locations?

Reply:
Yes, the local variables (`visible` and `rect`) store the visibility state and window rectangle (size/position) respectively while we remove/re-create the controls widgets